src/ChatPage.cpp

```diff
@@ -324,9 +326,27 @@ ChatPage::ChatPage(QSharedPointer<UserSettings> userSettings, QWidget *parent)
                   }
 
                   QSize dimensions;
-                  if (mimeClass == "image")
+                  QString blurhash;
+                  if (mimeClass == "image") {
                           dimensions = QImageReader(dev.data()).size();
 
+                          QImage img;
+                          img.loadFromData(bin);
+                          if (img.height() > 200 && img.width() > 360)
+                                  img = img.scaled(360, 200, Qt::KeepAspectRatioByExpanding);
+                          std::vector<unsigned char> data;
+                          for (int y = 0; y < img.height(); y++) {
+                                  for (int x = 0; x < img.width(); x++) {
+                                          auto p = img.pixel(x, y);
+                                          data.push_back(static_cast<unsigned char>(qRed(p)));
+                                          data.push_back(static_cast<unsigned char>(qGreen(p)));
+                                          data.push_back(static_cast<unsigned char>(qBlue(p)));
+                                  }
+                          }
```

Comment:
I feel like this blurhash encoding prep work would belong best outside of chatpage.cpp from an organization perspective. then we can centralize the logic and re-use the encoding for videos or redacted stuff, or whatever else down the line.

Reply:
Well, currently this inside the upload function, that is common for every media. I think we need to factor a lot more of it out at some point, but I'd say that will be a separate PR? Like when video thumbnailing gets added?